week-3/stretch/rotate-char.js

```diff
@@ -6,6 +6,33 @@
 // This function is commonly used for weak text encryption and decryption,
 // where shifting characters by a certain value can obscure their meaning or reveal hidden messages.
 
+function isLetterOrNumber(char) {
+  return /^[a-zA-Z0-9]+$/.test(char);
+}
+
+function rotateCharacter(char, shift) {
+  if (isLetterOrNumber(char) && char.length == 1 && typeof shift == "number" && shift >= 0) {
```

Comment:
This is very minor, but with short circuit evaluation (https://en.wikipedia.org/wiki/Short-circuit_evaluation) it's marginally preferable to put 'cheap' operations first. `char.length === 1` is computationally easier than regex pattern matching, so you'd have slightly better performance by putting that condition check first. If it's false,  isLetterOrNumber will never be called. Here, honestly, it really doesn't matter. But being aware of the short circuit evaluation might help you later where it is more important.

Also try to use strict equality (`===`) where you can

Reply:
Nice topic! Thank you. I think in more complex apps that demand more computational power, it will be really important.